File: GXpert/GXpert.Web/Modules/Analytics/ExamAttemptQuestion/ExamAttemptQuestionColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Analytics.Columns;

[ColumnsScript("Analytics.ExamAttemptQuestion")]
[BasedOnRow(typeof(ExamAttemptQuestionRow), CheckNames = true)]
public class ExamAttemptQuestionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ExamAttemptStudentAnswerSheetUpload { get; set; }
    public string ExamQuestionRightAnswer { get; set; }
    [EditLink]
    public string OptionSelected { get; set; }
    public int MarksObtained { get; set; }
    [QuickFilter]
    public short Attemptstatus { get; set; }
    public int OutOfmarks { get; set; }
    public string Result { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Analytics/ExamAttemptQuestionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ExamAttemptQuestionRow } from "./ExamAttemptQuestionRow";

export interface ExamAttemptQuestionColumns {
    Id: Column<ExamAttemptQuestionRow>;
    ExamAttemptStudentAnswerSheetUpload: Column<ExamAttemptQuestionRow>;
    ExamQuestionRightAnswer: Column<ExamAttemptQuestionRow>;
    OptionSelected: Column<ExamAttemptQuestionRow>;
    MarksObtained: Column<ExamAttemptQuestionRow>;
    Attemptstatus: Column<ExamAttemptQuestionRow>;
    OutOfmarks: Column<ExamAttemptQuestionRow>;
    Result: Column<ExamAttemptQuestionRow>;
}

export class ExamAttemptQuestionColumns extends ColumnsBase<ExamAttemptQuestionRow> {
    static readonly columnsKey = 'Analytics.ExamAttemptQuestion';
    static readonly Fields = fieldsProxy<ExamAttemptQuestionColumns>();
}